### PR TITLE
Feature/combine quantize

### PR DESCRIPTION
**Pull request by xhuohai**
1. combine quantize with reshape/transpose
2. dump ir if CheckedDataType/CheckedShape fail

### CODE REVIEW

**Review by zhen8838 — APPROVED** [review]

File: src/Nncase.Tests/Rules/Neutral/UnitTestCombineQuantize.cs [hunk 1]
```diff
@@ -94,4 +107,88 @@ public async Task TestCombineQuantizeConcatNegative()
...
+        var preValue = CompilerServices.Evaluate(rootPre, feedDict);
+
+        var analysis = new Dictionary<Type, IAnalysisResult> { [typeof(IExprUserAnalysisResult)] = AnalyzerMananger.GetAnaylsis<IExprUserAnalysisResult>(rootPre), };
+
+        var pass = new DataflowPass();
+        pass.Add<CombineQuantizeReshape>();
+        var rootPost = (Function)await pass.RunAsync(rootPre, new() { AnalysisResults = analysis });
+        Assert.NotEqual(preHashCode, rootPost.GetHashCode());
+
+        Assert.Equal(preValue, CompilerServices.Evaluate(rootPost, feedDict));
```
Comment (sunnycase) [inline src/Nncase.Tests/Rules/Neutral/UnitTestCombineQuantize.cs:133]: Use `TestMatched`.

File: src/Nncase.Core/IR/Expr.cs [hunk 2]
```diff
@@ -85,20 +86,38 @@ public IRType CheckedType
...
-        _ => throw new InvalidOperationException("Only The Expr Have CheckedType Can Get It's Shape"),
-    };
+        get
+        {
+            switch (CheckedType)
+            {
+                case TensorType type:
+                    return type.Shape;
+                default:
+                    DumpScope.Current.DumpIR(this, "checked_shape_error");
```
Comment (sunnycase) [inline src/Nncase.Core/IR/Expr.cs:98]: Use Pascal casing.

File: src/Nncase.Core/IR/Expr.cs [hunk 3]
```diff
@@ -85,20 +86,38 @@ public IRType CheckedType
...
-        _ => throw new InvalidOperationException("Only The Expr Have CheckedType Can Get It's Shape"),
-    };
+        get
+        {
+            switch (CheckedType)
+            {
+                case TensorType type:
+                    return type.Shape;
+                default:
+                    DumpScope.Current.DumpIR(this, "CheckedShapeError");
```
Comment (sunnycase) [inline src/Nncase.Core/IR/Expr.cs:98]: Wrap the dump with a flag, it's not necessary to dump the expr in every situation.

Comment (xhuohai) [reply src/Nncase.Core/IR/Expr.cs]: changed.

**Review by sunnycase — APPROVED** [review]

**Pull request merged**